erpnext/controllers/buying_controller.py

```diff
@@ -488,6 +493,39 @@ def update_stock_ledger(self, allow_negative_stock=False, via_landed_cost_vouche
 						"serial_no": cstr(d.rejected_serial_no).strip(),
 						"incoming_rate": 0.0
 					}))
+			elif has_product_bundle(d.item_code):
+				for p in self.get("packed_items"):
+					if p.parent_detail_docname == d.name and p.parent_item == d.item_code:
+						if p.qty:
+							sle = self.get_sl_entries(p, {
+								"actual_qty": flt(p.qty),
+								"serial_no": cstr(p.serial_no).strip()
+							})
+							if self.is_return:
+								original_incoming_rate = frappe.db.get_value("Stock Ledger Entry",
+									{"voucher_type": "Purchase Receipt",
+									"voucher_no": self.return_against,
+									"item_code": p.item_code},
+									"incoming_rate")
+
+								sle.update({
+									"outgoing_rate": original_incoming_rate
+								})
+							else:
+								val_rate_db_precision = 6 if cint(self.precision("valuation_rate", d)) <= 6 else 9
+								incoming_rate = flt(p.rate, val_rate_db_precision)
+								sle.update({
+									"incoming_rate": incoming_rate
+								})
+							sl_entries.append(sle)
+
+						if flt(d.rejected_qty) != 0:
+							sl_entries.append(self.get_sl_entries(p, {
+								"warehouse": d.rejected_warehouse,
+								"actual_qty": flt(d.rejected_qty)*(flt(p.qty)/flt(d.qty)),
+								"serial_no": cstr(d.rejected_serial_no).strip(),
+								"incoming_rate": 0.0
+							}))
```

Comment:
sl entries for product bundle items can be commonify with non product bundle items. 
For example
```
items = frm.doc.items + packing items
for d in items:
    create the sl entries 
```

Reply:
I avoided that for various reasons

1. Need to change AccountsController - set_stock_items method is defined in AccountsController
```
stock_items = self.get_stock_items()

for d in self.get('items'):
	if d.item_code in stock_items and d.warehouse:
```

2. There are various fields in item that are needed for packed_item and are not copied
```
conversion_factor
rejected_warehouse
rejected_qty
```
these fields are calculated while making sl entries. these fields will be need to be added to PackedItem